### PR TITLE
NEW: RM#78113 Generate default error page

### PR DESCRIPTION
**Pull request by zhanglian77**

### CODE REVIEW

**Comment by zhanglian77** [comment]
Please review this merge request and https://gitlab.catalyst.net.nz/SilverStripe/security-development-lifecycle-tool/-/merge_requests/309 both for RM#78113

**Comment by elliot-sawyer** [comment]
We need to warn users on this project that HomePage and ErrorPage will no longer be created on dev/build. They will need to be published manually if this change is merged.  Update the README or create a [wiki](https://github.com/NZTA/sdlt-framework/wiki) entry about the change

File: src/Page/HomePage.php [hunk 1]
```diff
@@ -90,7 +85,12 @@ public function requireDefaultRecords()
             if (!SiteTree::get_by_link($defaultHomepage)) {
                 $homepage = new HomePage();
                 $homepage->Title = _t(__CLASS__.'.DEFAULTHOMETITLE', 'Home');
-                $homepage->Content = _t(__CLASS__.'.DEFAULTHOMECONTENT', '<p>Welcome to SilverStripe! This is the default homepage. You can edit this page by opening <a href="admin/">the CMS</a>.</p><p>You can now access the <a href="http://docs.silverstripe.org">developer documentation</a>, or begin the <a href="http://www.silverstripe.org/learn/lessons">SilverStripe lessons</a>.</p>');
+                $homepage->Content = _t(__CLASS__.'.DEFAULTHOMECONTENT',
```
Comment (roopamjain01) [inline src/Page/HomePage.php:88]: Can you please move the first parameter also in a new line.

File: src/Page/HomePage.php [hunk 2]
```diff
@@ -99,5 +99,14 @@ public function requireDefaultRecords()
...
         }
+        if (!(ErrorPage::get()->Count())) {
+            $errorpage = new ErrorPage();
+            $errorpage->Title = 'Error Page';
+            $errorpage->ErrorCode = 404;
+            $errorpage->write();
+            $errorpage->copyVersionToStage(Versioned::DRAFT, Versioned::LIVE);
+            $errorpage->flushCache();
+            DB::alteration_message('Default Error page created', 'created');
+        }
```
Comment (roopamjain01) [inline src/Page/HomePage.php:110]: Could you please add an error page for error code 500 as well?

File: src/Page/HomePage.php (hunk 1 above)
Comment (zhanglian77) [reply src/Page/HomePage.php]: done

File: src/Page/HomePage.php (hunk 2 above)
Comment (zhanglian77) [reply src/Page/HomePage.php:118]: done

File: src/Page/HomePage.php [hunk 3]
```diff
@@ -99,5 +100,21 @@ public function requireDefaultRecords()
...
         }
+        if (!(ErrorPage::get()->Count())) {
+            $errorPageNotFound = new ErrorPage();
+            $errorPageNotFound->Title = '404 Error Page';
+            $errorPageNotFound->ErrorCode = 404;
+            $errorPageNotFound->write();
+            $errorPageNotFound->copyVersionToStage(Versioned::DRAFT, Versioned::LIVE);
+            $errorPageNotFound->flushCache();
+            DB::alteration_message('Default 404 Error page created', 'created');
+            $errorPageServerError = new ErrorPage();
```
Comment (roopamjain01) [inline src/Page/HomePage.php:111]: Please add a new empty line between the code(line number 110-111)  for readability and please add a comment as well before both the error page.

**Pull request merged**